src/doc.cpp

```diff
@@ -1538,13 +1538,15 @@ int Doc::PrepareLyricsEnd(FunctorParams *functorParams)
 {
     PrepareLyricsParams *params = dynamic_cast<PrepareLyricsParams *>(functorParams);
     assert(params);
-
-    if ((params->m_currentSyl && params->m_lastNote) && (params->m_currentSyl->GetStart() != params->m_lastNote)) {
+    if (!params->m_currentSyl) {
+        return FUNCTOR_STOP; // early return
+    }
+    if (params->m_lastNote && params->m_currentSyl->GetStart() != params->m_lastNote) {
         params->m_currentSyl->SetEnd(params->m_lastNote);
     }
     else if (m_options->m_openControlEvents.GetValue()) {
-        if ((params->m_currentSyl->GetWordpos() == sylLog_WORDPOS_i)
-            || (params->m_currentSyl->GetWordpos() == sylLog_WORDPOS_m)) {
+        sylLog_WORDPOS wordpos = params->m_currentSyl->GetWordpos();
+        if (wordpos == sylLog_WORDPOS_i || wordpos == sylLog_WORDPOS_m) {
```

Comment:
Sorry, one last thing I am noticing: please put both tests in ()

Reply:
https://en.cppreference.com/w/c/language/operator_precedence
Why are these extra parentheses needed?

Reply:
Are you offering me a C++ course? ;-)
It is a habit and prefer to have the code consistent. Thanks

Reply:
Well, your game - your rules.
As for me, parentheses which don't change natural operations precedence just create visual garbage. They might be useful in complex conditions. But rare in one-liners.
Pushed change.

Reply:
> As for me, parentheses which don't change natural operations precedence just create visual garbage.

Absolutely not.

Reply:
Ok. Different people - different opinions.
I have returned parentheses back.